### PR TITLE
chore: add hosted proxy

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Towards https://github.com/PostHog/product-internal/issues/602

## Changes

Add hosted proxy as pay gate

### CODE REVIEW

File: frontend/src/scenes/settings/project/Proxy.tsx [hunk 1]
```diff
@@ -88,23 +87,29 @@ export function Proxy(): JSX.Element {
...
-                dataSource={proxyRecords}
-                expandable={{
-                    expandedRowRender: (record) => <ExpandedRow record={record} />,
-                }}
-            />
-            {formState === 'collapsed' ? (
-                <LemonButton onClick={showForm} type="secondary" icon={<IconPlus />}>
-                    Add domain
-                </LemonButton>
+        <PayGateMini feature={AvailableFeature.HOSTED_REVERSE_PROXY}>
```
Comment (zlwaterfield) [inline frontend/src/scenes/settings/project/Proxy.tsx:90]: Should this `isCloudOrDev` check be outside of the paygate? I feel like non-cloud users should see the warning always.

Comment (daibhin) [reply frontend/src/scenes/settings/project/Proxy.tsx:90]: Good call, updated

File: frontend/src/scenes/settings/project/HostedReverseProxy.tsx [hunk 2]
```diff
@@ -87,24 +86,28 @@ export function Proxy(): JSX.Element {
...
+                    <LemonButton onClick={showForm} type="secondary" icon={<IconPlus />}>
+                        Add domain
+                    </LemonButton>
+                ) : (
+                    <CreateRecordForm />
+                )}
+            </div>
+        </PayGateMini>
+    ) : (
+        <LemonBanner type="warning">Using a reverse proxy only works in PostHog Cloud</LemonBanner>
```
Comment (raquelmsmith) [inline frontend/src/scenes/settings/project/HostedReverseProxy.tsx:110]: They paygate actually has CTAs for posthog cloud, so I think this check is unnecessary.

File: frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png
[the remark this answers is not in the document]
Comment (daibhin) [reply frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png:1]: Will the billing changes need to be merged first?

[the remark this answers is not in the document]
Comment (raquelmsmith) [reply frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png:1]: No, these are based on fixtures. I'll fix

[the remark this answers is not in the document]
Comment (raquelmsmith) [reply frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png:1]: No, these are based on fixtures. I'll fix

[the remark this answers is not in the document]
Comment (daibhin) [reply frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png:1]: Thank you :)

[the remark this answers is not in the document]
Comment (raquelmsmith) [reply frontend/__snapshots__/scenes-other-settings--settings-organization--dark.png:1]: Done!

**Review by frankh — APPROVED** [review]
🚀

File: posthog/api/proxy_record.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def create(self, request, *args, **kwargs):
         )
 
         temporal = sync_connect()
-        inputs = CreateHostedProxyInputs(
+        inputs = CreateManagedProxyInputs(
```
Comment (daibhin) [inline posthog/api/proxy_record.py:61]: @frankh just making sure that it's ok to rename all the temporal stuff to match the new name of the feature. Hoping there's no config elsewhere that references "Hosted"

Comment (frankh) [reply posthog/api/proxy_record.py:61]: wait no nevermind this is all in the main app

**Pull request merged**